fpop/cp2k.py

```diff
@@ -0,0 +1,225 @@
+        with open('CELL_PARAMETER', 'w') as file:
+            file.write(f"A {cell_params[0,0]:14.8f} {cell_params[0,1]:14.8f} {cell_params[0,2]:14.8f}\n")
+            file.write(f"B {cell_params[1,0]:14.8f} {cell_params[1,1]:14.8f} {cell_params[1,2]:14.8f}\n")
+            file.write(f"C {cell_params[2,0]:14.8f} {cell_params[2,1]:14.8f} {cell_params[2,2]:14.8f}\n")
+
+        # Write the CP2K input file content
+        Path('input.inp').write_text(cp2k_inputs.inp_template)
+
+        # Copy optional files to the working directory
+        if optional_artifact:
+            for file_name, file_path in optional_artifact.items():
+                content = file_path.read_text()
+                Path(file_name).write_text(content)
+
+
+class RunCp2k(RunFp):
+    def input_files(self, task_path) -> List[str]:
+        """
+        The mandatory input files to run a CP2K task.
+        
+        Returns
+        -------
+        files: List[str]
+            A list of mandatory input file names.
+        """
+        return ["input.inp", "CELL_PARAMETER", "coord.xyz"]
+
+    def run_task(
+        self,
+        backward_dir_name,
+        log_name,
+        backward_list: List[str],
+        run_image_config: Optional[Dict] = None,
+        optional_input: Optional[Dict] = None,
+    ) -> str:
+        """
+        Defines how one FP task runs.
+
+        Parameters
+        ----------
+        backward_dir_name : str
+            The name of the directory which contains the backward files.
+        log_name : str
+            The name of log file.
+        backward_list : List[str]
+            The output files the users need. For example: ["output.log", "trajectory.xyz"]
+        run_image_config : Dict, optional
+            Keyword args defined by the developer. For example:
+            {
+              "command": "source /opt/intel/oneapi/setvars.sh && mpirun -n 64 /opt/cp2k/bin/cp2k.popt"
+            }
+        optional_input : Dict, optional
+            The parameters developers need in runtime. For example:
+            {
+                "conf_format": "cp2k/input"
+            }
+        
+        Returns
+        -------
+        backward_dir_name : str
+            The directory name which contains the files users need.
+        """
+        # Get the run command
+        if run_image_config:
+            command = run_image_config.get("command")
+            if not command:
+                raise ValueError("Command not specified in run_image_config")
+        else:
+            raise ValueError("run_image_config is missing")
+        
+        # Run CP2K command and write output to log file
+        command = " ".join([command, ">", log_name])
+        kwargs = {"try_bash": True, "shell": True}
+        if run_image_config:
+            kwargs.update(run_image_config)
+            kwargs.pop("command", None)
+        
+        # Execute command
+        ret, out, err = run_command(command, raise_error=False, **kwargs)
```

Comment:
Change to `ret, out, err = run_command(command, raise_error=False, **kwargs)  # type: ignore` can solve type checker error.